homeassistant/components/rachio/coordinator.py

```diff
@@ -54,3 +67,54 @@ async def _async_update_data(self) -> dict[str, Any]:
         except Timeout as err:
             raise UpdateFailed(f"Could not connect to the Rachio API: {err}") from err
         return {valve[KEY_ID]: valve for valve in data[1][KEY_VALVES]}
+
+
+class RachioScheduleUpdateCoordinator(DataUpdateCoordinator[list[dict[str, Any]]]):
+    """Coordinator for fetching hose timer schedules."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        rachio: Rachio,
+        base_station,
+    ) -> None:
+        """Initialize a Rachio schedule coordinator."""
+        self.hass = hass
+        self.rachio = rachio
+        self.base_station = base_station
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN} schedule update coordinator",
+            update_interval=timedelta(minutes=30),
+        )
+
+    async def _async_update_data(self) -> list[dict[str, Any]]:
+        """Retrieve data for the past week and the next 60 days."""
+        _time_start: datetime = dt_util.now() - timedelta(days=7)
+        _time_end: datetime = dt_util.now() + timedelta(days=60)
```

Comment:
Good question. I don't know that it couldn't be UTC, but I guess I'd prefer it be local for events that fall on the edge of the time frames that might be omitted otherwise. For example an event 7 days ago where the current UTC time has moved to the next day would be left out, if I'm thinking about that correctly.